src/internal/connector/support/m365Transform.go

```diff
@@ -88,6 +88,14 @@ func ToEventSimplified(orig models.Eventable) models.Eventable {
 	orig.SetICalUId(nil)
 	orig.SetId(nil)
 
+	// Sanitize recurrence timezone.
+	if orig.GetRecurrence() != nil {
+		recurrenceTimezone := orig.GetRecurrence().GetRange().GetRecurrenceTimeZone()
+		if recurrenceTimezone != nil && len(*recurrenceTimezone) == 0 {
+			orig.GetRecurrence().GetRange().SetRecurrenceTimeZone(nil)
+		}
```

Comment:
We now have helper funcs for pointer values like this:
```suggestion
		recurrenceTimezone := ptr.Val(orig.GetRecurrence().GetRange().GetRecurrenceTimeZone())
		if len(recurrenceTimezone) == 0 {
			orig.GetRecurrence().GetRange().SetRecurrenceTimeZone(nil)
		}
```